core-bundle/src/Resources/views/Frontend/preview_toolbar_base_js.html.twig

```diff
@@ -299,5 +299,26 @@
                 });
             });
         });
+
+        // copy published link to clipboard
+        window.addEventListener('cto-toolbar-loaded', function() {
+            const copyButton = document.getElementById('copyPublishedPath');
+            copyButton.addEventListener('click', function(event) {
+                const input = document.createElement('input');
+                input.value = window.location.href.replace('{{ preview_script }}', '');
+                function clipboardFallback() {
+                    input.select();
+                    input.setSelectionRange(0, 99999);
+                    document.execCommand('copy');
+                }
+                event.preventDefault();
+                if (navigator.clipboard && navigator.clipboard.writeText) {
+                    navigator.clipboard.writeText(input.value).catch(clipboardFallback);
```

Comment:
```suggestion
                const href = window.location.href.replace('{{ preview_script }}', '');
                function clipboardFallback() {
                    const input = document.createElement('input');
                    input.value = href;
                    document.body.appendChild(input);
                    input.select();
                    input.setSelectionRange(0, 99999);
                    document.execCommand('copy');
                    document.body.removeChild(input);
                }
                event.preventDefault();
                if (navigator.clipboard && navigator.clipboard.writeText) {
                    navigator.clipboard.writeText(href).catch(clipboardFallback);
```

I think the input has to be injected into the document in order for the `copy` command to work.

Reply:
@ausi true! I just implemented that. Also tested with Safari and "async clipboard API" disabled.